theseus/optimizer/manifold_gaussian.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from itertools import count
+from typing import List, Optional, Sequence, Tuple
+
+import torch
+
+from theseus.geometry import LieGroup, Manifold
+
+
+class ManifoldGaussian:
+    _ids = count(0)
+
+    def __init__(
+        self,
+        mean: Sequence[Manifold],
+        precision: Optional[torch.Tensor] = None,
+        name: Optional[str] = None,
+    ):
+        self._id = next(ManifoldGaussian._ids)
+        if name is None:
+            name = f"{self.__class__.__name__}__{self._id}"
+        self.name = name
+
+        dof = 0
+        for v in mean:
+            dof += v.dof()
+        self._dof = dof
+
+        self.mean = mean
+        if precision is None:
+            precision = torch.zeros(mean[0].shape[0], self.dof, self.dof).to(
```

Comment:
Good catch! Zero default would't be invertible and could cause problems.

Reply:
I had initialised the precision with zeros because then I can directly use it for messages which are initialised with 0 precision (i.e. zero information). But I agree that for wider use it makes more sense to initialise with the identity.